sys/cddl/contrib/opensolaris/uts/common/sys/avl_impl.h

```diff
@@ -101,42 +101,25 @@ struct avl_node {
  *
  * pointer to the parent of the current node is the high order bits
  */
-#ifndef __CHERI_PURE_CAPABILITY__
 #define	AVL_XPARENT(n)	((struct avl_node *)((n)->avl_pcb & (uintptr_t)~7))
 #define	AVL_SETPARENT(n, p)						\
 	((n)->avl_pcb = (((n)->avl_pcb & (uintptr_t)7) | (uintptr_t)(p)))
```

Comment:
Will this infer provenance correctly?  It seems like it needs to come from `p` (the real pointer) with the low 3 bits coming from the existing `avl_pcb`?  (The old code doesn't seem to do that either though...)

Reply:
Looks like it doesn't: https://cheri-compiler-explorer.cl.cam.ac.uk/z/NRAGLy

I might make the ambigious warning an error by default that you can turn off with -Wno-error.